File: go/prime-factors/prime_factors.go
package primefactors

func Factors(n int64) []int64 {
	var primeFactors []int64
	var potentialFactor int64 = 2
	for n > 1 {
		if n%potentialFactor == 0 {
			primeFactors = append(primeFactors, potentialFactor)
			n /= potentialFactor
		} else {
			potentialFactor++
		}
	}
	return primeFactors
}

File: php/prime-factors/PrimeFactors.php
<?php

declare(strict_types=1);

function factors(int $number): array
{
	$prime_factors = [];
	$potential_factor = 2;
	while($number > 1) {
		if ($number % $potential_factor == 0) {
			$prime_factors[] = $potential_factor;
			$number /= $potential_factor;
		} else {
			$potential_factor++;
		}
	}
	return $prime_factors;
}
